python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -3161,7 +3166,7 @@ class DPExplainableBoostingRegressor(EBMModel, RegressorMixin, ExplainerMixin):
             - `'nominal'`: Categorical where the order has no meaning. Eg: country names. Uses private categorical binning.
     max_bins : int, default=32
         Max number of bins per feature.
-    exclude : list of tuples of feature indices|names, default=[]
+    exclude : list of tuples of feature indices|names, default=None
```

Comment:
I used an empty array by default here because I felt it was more self-documenting.  If the default is None then you don't know without looking at the documentation what kinds of inputs are expected.  Does a string work?  Does an integer?  Having the empty list at least documents that we're expecting a list of several items.  I'm happy to change this to None though if there's some big issue with scikit-learn compliance or something similar.  The self-documentation aspect is really only minimally useful.

Reply:
I agree that the empty list is a bit more instructive. On the other hand, your type hint `exclude: Optional[Sequence[Union[int, str, Sequence[Union[int, str]]]]] = []` suggests that `None` is a valid value. `Optional` explicitly means `or None`. On the other hand, `exclude=None` is very directly understandable and instructive.

I think the main reason, scikit-learn doesn't allow `exclude=[]`, is that mutable default values are considered extremely harmful in Python. If someone were to modify `exclude`, all following function calls would suddenly have a magically different default values. That's why `None` is the established default value in the Python community. An empty list is very surprising and considered bad practice. Things like `dataclasses` handle this by providing factories instead of mutable default arguments.

Reply:
Ok, sounds good.  None it is then.